app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt

```diff
@@ -213,34 +214,66 @@ class TrackingProtectionFragment : PreferenceFragmentCompat() {
 
         customCookies.onPreferenceChangeListener = object : SharedPreferenceUpdater() {
             override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
-                    updateTrackingProtectionPolicy()
                 customCookiesSelect.isVisible = !customCookies.isChecked
-                return super.onPreferenceChange(preference, newValue)
+                return super.onPreferenceChange(preference, newValue).also {
+                    updateTrackingProtectionPolicy()
```

Comment:
Since `updateTrackingProtectionPolicy` uses information from shared prefs, it's important that we call super (and write to SP) before calling it.